src/test/java/com/wikia/webdriver/elements/mercury/components/Search.java

```diff
@@ -86,4 +78,70 @@ public Search clickClearSearchButton() {
     return this;
   }
 
+  public SearchResultsPage clickEnterAndNavigateToSearchResults() {
+    new Actions(driver).sendKeys(Keys.ENTER).perform();
+
+    return new SearchResultsPage();
+  }
+
+  public boolean isInputFieldSearchIconVisible() {
+    return isElementVisible(inputFieldSearchIcon);
+  }
+
+  public boolean isClearSearchButtonVisible() {
+    return isElementVisible(clearSearchButton);
+  }
+
+  public boolean isSearchInputFieldVisible() {
+    return isElementVisible(searchInput);
+  }
+
+  public boolean isSearchInputFieldEditable() {
+    try {
+      wait.forElementClickable(searchInput, 0).click();
+      return true;
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.logInfo(e.getMessage());
+      return false;
+    } catch (WebDriverException e) {
+      // @TODO: this would be best pushed to Selenium upstream to actually throw a more specific
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1135)